### PR TITLE
binance/perf: Optimize UFuturesNewOrder function parameters

### PR DESCRIPTION
**Pull request by xiiiew**
# PR Description

```go
// old version
func (b *Binance) UFuturesNewOrder(ctx context.Context, symbol currency.Pair, side, positionSide, orderType, timeInForce,
	newClientOrderID, closePosition, workingType, newOrderRespType string,
	quantity, price, stopPrice, activationPrice, callbackRate float64, reduceOnly bool)(UOrderData, error) {
}

// new version
func (b *Binance) UFuturesNewOrder(ctx context.Context, data *UFuturesNewOrderRequest)(UOrderData, error) {
}

type UFuturesNewOrderRequest struct {
	Symbol           currency.Pair `json:"symbol"`
	Side             string        `json:"side"`
	PositionSide     string        `json:"position_side"`
	OrderType        string        `json:"order_type"`
	TimeInForce      string        `json:"time_in_force"`
	NewClientOrderID string        `json:"new_client_order_id"`
	ClosePosition    string        `json:"close_position"`
	WorkingType      string        `json:"working_type"`
	NewOrderRespType string        `json:"new_order_resp_type"`
	Quantity         float64       `json:"quantity"`
	Price            float64       `json:"price"`
	StopPrice        float64       `json:"stop_price"`
	ActivationPrice  float64       `json:"activation_price"`
	CallbackRate     float64       `json:"callback_rate"`
	ReduceOnly       bool          `json:"reduce_only"`
}
```

UFuturesNewOrder function parameter is too long to be convenient to call. I modified the function to require a struct and all parameters are in it.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance_test.go [hunk 1]
```diff
@@ -425,7 +425,25 @@ func TestUFuturesNewOrder(t *testing.T) {
...
+			ClosePosition:    "",
+			WorkingType:      "",
+			NewOrderRespType: "",
+			Quantity:         1,
+			Price:            1,
+			StopPrice:        0,
+			ActivationPrice:  0,
+			CallbackRate:     0,
+			ReduceOnly:       false,
+		},
```
Comment (shazbert) [inline exchanges/binance/binance_test.go:445]: We can remove the zero values:

```suggestion
		&UFuturesNewOrderRequest{
			Symbol:           currency.NewPair(currency.BTC, currency.USDT),
			Side:             "BUY",
			OrderType:        "LIMIT",
			TimeInForce:      "GTC",
			Quantity:         1,
			Price:            1,
		},
```

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks for opening this PR, this is much better! Just a suggestion and there is a linter issue in `ufutures_types` files.

```
exchanges\binance\ufutures_types.go:4: File is not `gci`-ed with --skip-generated -s standard,default (gci)
	"github.com/thrasher-corp/gocryptotrader/currency"
exchanges\binance\ufutures_types.go:5: File is not `gci`-ed with --skip-generated -s standard,default (gci)
	"time"
```

File: exchanges/binance/binance_test.go (hunk 1 above)
Comment (xiiiew) [reply exchanges/binance/binance_test.go]: Thanks for your suggestion

**Comment by thrasher-** [comment]
Thanks for this PR @xiiiew ! Just needs some linters fixed and then this looks good to go. Patch file can be found here: https://gist.github.com/thrasher-/6c4e33012646c8a8cd99e2a0b29e002f

**Review by thrasher- — CHANGES_REQUESTED** [review]
Only one more linter issue remaining now which gofmt is picking up ^^

**Comment by xiiiew** [comment]
> Thanks for this PR @xiiiew ! Just needs some linters fixed and then this looks good to go. Patch file can be found here: https://gist.github.com/thrasher-/6c4e33012646c8a8cd99e2a0b29e002f

Thanks for your reply, I have fixed this issue.

**Review by gloriousCode — APPROVED** [review]
tACK! Thank you for your contribution. Much nicer :rainbow:

**Review by thrasher- — APPROVED** [review]
Thank you for making those changes!

**Review by shazbert — APPROVED** [review]
looking good! tAck

```
Binance attempt 1 request path: https://fapi.binance.com/fapi/v1/order?price=1&quantity=1000000000&recvWindow=5000&side=BUY&symbol=LTCBTC&timeInForce=GTC&timestamp=1663728484904&type=LIMIT&signature=0ec5b7f0369b968708a8becfb42b2e360eae4bbf4d17646ba6808300b26effa7
Binance request header [X-Mbx-Apikey]: [llollcat]
Binance request type: POST
```

**Comment by xiiiew** [comment]
Thanks you all, this change will make development easier

**Pull request merged**